### PR TITLE
Signed Integers `TwosComplement` fix to `WrappingNeg`

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Bug fix

## Changes

The following changes have been made:

- The `TwosComplement` trait was not correctly implemented. To stay in line with Rust's practices, the `WrappingNeg` trait now replaces it with the original desired functionality of the `TwosComplement` trait. 
- Bits for `I128` and `I256` are now correct
- The indent for `I128` and `I256` are now correct

## Notes

- This PR changes the folder names in the tests from `iX_twos_complement_test` to `iX_wrapping_neg_test`.

## Breaking Changes

The following demonstrates the breaking change. While this example code uses the `I8` type, the same logic may be applied to the `I16`, `I32`, `I64`, `I128`, and `I256` types. 

Before:
```sway
let my_i8 = i8::zero();
let twos_complement = my_i8.twos_complement();
```

After:
```sway
let my_i8 = i8::zero();
let wrapping_neg = my_i8.wrapping_neg();
```

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.
- [x] I have updated the changelog to reflect the changes on this PR.

### CODE REVIEW

File: libs/src/signed_integers/common.sw [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-/// Trait for the Two's Complement of a value.
-pub trait TwosComplement {
-    /// Returns the two's complement of a value.
+/// Wrapping (modular) negation. Computes -self, wrapping around at the boundary of the type.
+pub trait WrappingNeg {
+    /// Negates a signed number.
+    ///
+    /// # Additional Information
+    ///
+    /// * The only case where such wrapping can occur is when one negates self::min(). In such a case, this function returns self::min() itself.
```
Comment (SwayStar123) [inline libs/src/signed_integers/common.sw:9]: Not sure i understand this, shouldnt -(self::min) return self::max?

Comment (SwayStar123) [reply libs/src/signed_integers/common.sw:9]: Trying 

`fn main () {
    println!("{}", -i8::MIN)
}`
On returns a compiler error. Not sure if returning self.min is the right behavior here, would prefer a result or a revert

Comment (K1-R1) [reply libs/src/signed_integers/common.sw:9]: This is correct behaviour for [wrapping_neg](https://doc.rust-lang.org/std/primitive.i32.html#method.wrapping_neg).

Although I agree that [checked_neg](https://doc.rust-lang.org/std/primitive.i32.html#method.checked_neg) is less user error-prone.

Comment (bitzoic) [reply libs/src/signed_integers/common.sw:9]: No max value is 2 ^ 7 - 1, min value is - 2 ^ 7

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,8 +21,23 @@ Description of the upcoming release here.
 ### Fixed
 
 - [#258](https://github.com/FuelLabs/sway-libs/pull/258) Fixes incorrect instructions on how to run tests in README and docs hub.
+- [#263](https://github.com/FuelLabs/sway-libs/pull/263) Fixes `I256`'s returned bits.
```
Comment (SwayStar123) [inline CHANGELOG.md:24]: Do you need two rows for these?

**Review by SwayStar123 — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (bitzoic) [reply CHANGELOG.md:24]: Yes it's best practice when there are multiple changes in a single PR.

**Review by K1-R1 — APPROVED** [review]

**Comment by bitzoic** [comment]
Dependent on https://github.com/FuelLabs/docs-hub/pull/340

**Review by SwayStar123 — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**